### PR TITLE
Native splashscreen

### LINKED ISSUE #247: Duplicate splash screen
Beginning with Android 11 or Android 12 every app you open shows a native splashscreen.
Additionally this app currently shows an own splashscreen, we could merge those two into one native one.

Here's the problem (on Android 13):

https://user-images.githubusercontent.com/46033730/194776278-8111af1a-5a92-4312-b44d-31198bf3f051.mp4

### PR DESCRIPTION
**Pull request by cyb3rko**
Let's go, the next PR.
This one is a small one. The implementation already seems to work for me.

Do you find any edge cases, where it does not work?
Hopefully I will be able to test this on older devices in the next days to see how Android 11 and below behave.
So again, please don't merge yet.

Fixes #247

### CODE REVIEW

File: app/src/main/kotlin/com/github/gotify/init/InitializationActivity.kt [hunk 1]
```diff
@@ -106,6 +112,7 @@ internal class InitializationActivity : AppCompatActivity() {
 
         settings.setUser(user.name, user.isAdmin)
         requestVersion {
+            splashScreenActive = false
```
Comment (jmattheis) [inline app/src/main/kotlin/com/github/gotify/init/InitializationActivity.kt:115]: This Closure in requestVersion doesn't run on the main thread, could there be thread safety problems with the splashScreenActive variable?

File: app/build.gradle [hunk 2]
```diff
@@ -65,6 +65,7 @@ if (project.hasProperty('sign')) {
 dependencies {
```
Comment (jmattheis) [inline app/build.gradle:65]: The first time I started the app with these changes, it only showed the splashscreen and didn't continue further. I sadly couldn't reproduce it afterwards. Do you have an idea what could've caused this?

File: app/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -17,12 +17,12 @@
         android:supportsRtl="true"
         android:usesCleartextTraffic="true"
         android:networkSecurityConfig="@xml/network_security_config"
-        android:theme="@style/AppTheme">
+        android:theme="@style/AppTheme.SplashScreen">
```
Comment (jmattheis) [inline app/src/main/AndroidManifest.xml:20]: If the user has selected the dark theme then the splash screen still has a light background.
[Peek 2023-01-31 18-21.webm](https://user-images.githubusercontent.com/14895212/215835615-4a4bcbcb-245d-4563-856d-55ea2cbb1027.webm)

Comment (cyb3rko) [reply app/src/main/AndroidManifest.xml:20]: That's what I thought, didn't have the time to test it yet.

What about the Android version? Which API level are you running this on?

File: app/build.gradle (hunk 2 above)
Comment (cyb3rko) [reply app/build.gradle:65]: Are you using an emulator?

Comment (jmattheis) [reply app/build.gradle:65]: Yes.

File: app/src/main/AndroidManifest.xml (hunk 3 above)
Comment (jmattheis) [reply app/src/main/AndroidManifest.xml:20]: I've tested it with 25 and 33 not sure which on which I tested this.

File: app/build.gradle (hunk 2 above)
Comment (cyb3rko) [reply app/build.gradle:65]: Maybe that's the problem. This library has some problems being launched on an emulator from Android Studio.
If you close the app and start it from the UI on the emulator again it should work (if that's really the problem).

Comment (jmattheis) [reply app/build.gradle:65]: Yeah, probably a good enough explanation for now.

File: app/src/main/AndroidManifest.xml (hunk 3 above)
Comment (cyb3rko) [reply app/src/main/AndroidManifest.xml:20]: I see, have you tested it on a physical device as well?
I suppose you would not have that problem there. But no guarantee.

Comment (cyb3rko) [reply app/src/main/AndroidManifest.xml:20]: I will take a look. You can change the background color of the SplashScreen, but it is pre-defined in xml and not dynamically changeable I think.

File: app/src/main/kotlin/com/github/gotify/init/InitializationActivity.kt (hunk 1 above)
Comment (cyb3rko) [reply app/src/main/kotlin/com/github/gotify/init/InitializationActivity.kt:115]: I'm not sure there.
As long as that variable is false, the splashScreen will stay active.
And as long as the variable at some point jumps to true, the splashScreen will disappear.

If there could be cases that the variable does not get changed, tbh I don't know.

File: app/build.gradle (hunk 2 above)
Comment (cyb3rko) [reply app/build.gradle:65]: Have you tested it on a physical device as well?
There you should not have that problem.

Comment (jmattheis) [reply app/build.gradle:65]: No, haven't tested it on a physical device yet. But I couldn't reproduce the problem afterward, so it was hopefully something cached related from the emulator.

File: app/src/main/AndroidManifest.xml (hunk 3 above)
Comment (cyb3rko) [reply app/src/main/AndroidManifest.xml:20]: After some investigation I found out that this is the expected behaviour.
The native splashscreen background seems to adapt to your system light/dark mode.
Everything afterwards of course follows the in-app setting.

Google Maps and Google Chrome work like this as well.
See here (system dark mode, in-app light mode):

https://user-images.githubusercontent.com/46033730/216186950-ca1c63fb-4d25-49b1-b35d-37538edf9967.mp4

File: app/build.gradle (hunk 2 above)
Comment (cyb3rko) [reply app/build.gradle:65]: I've tested it on two Emulators now (APIs 25 and 33 as well). Everything worked fine, on one device it took a second or two to do something because it had just booted up, but both showed me the login page.

File: app/src/main/AndroidManifest.xml (hunk 3 above)
Comment (jmattheis) [reply app/src/main/AndroidManifest.xml:20]: Thanks for investigating, the behavior is okay then.

File: app/src/main/kotlin/com/github/gotify/init/InitializationActivity.kt (hunk 1 above)
Comment (cyb3rko) [reply app/src/main/kotlin/com/github/gotify/init/InitializationActivity.kt:115]: Am I stupid or do we already run 'callInUI' in requestVersion?

---

image

Comment (jmattheis) [reply app/src/main/kotlin/com/github/gotify/init/InitializationActivity.kt:115]: Uff, sorry, I was pretty sure I only saw Callback.call there (:.

Comment (cyb3rko) [reply app/src/main/kotlin/com/github/gotify/init/InitializationActivity.kt:115]: Alright, no problem

File: app/build.gradle (hunk 2 above)
Comment (jmattheis) [reply app/build.gradle:65]: I've tested it on physical devices with Android 10 & 13, works fine.

**Review by jmattheis — APPROVED** [review]

**Comment by jmattheis** [comment]
@cyb3rko Do we have anything left that should be included in the next release?

**Comment by cyb3rko** [comment]
@jmattheis  I don't think so.
Everything seems to work now.

**Comment by jmattheis** [comment]
@cyb3rko Awesome!

**Comment by jmattheis** [comment]
FYI: I'll create a release tomorrow and check if the app keeps the websocket connection alive over the night.

**Comment by cyb3rko** [comment]
Good idea.

**Comment by cyb3rko** [comment]
@jmattheis And maybe you should change to hint at https://github.com/gotify/android#building to Java 11.

**Comment by cyb3rko** [comment]
Btw, are you interested in a simple design update?
I played around with Material 3 components as a replacement for the rather old fashioned design elements in this app.

What do you think?

**Comment by jmattheis** [comment]
@cyb3rko I've nothing against it, as long as the layout doesn't change too much (:.
